Core: Upgrade babel

hipstersmoothie opened the pull request:
## What I did

Upgrade babel and typescript in preparation for ESM builds.

This PR is mainly so I don't have to deal with merge conflicts anymore in my other PR.

## How to test

- Is this testable with Jest or Chromatic screenshots? no
- Does this need a new example in the kitchen sink apps? no
- Does this need an update to the documentation? no

hipstersmoothie committed:
upgrade typescript + babel

hipstersmoothie commented:
@shilman This PR should alleviate some of my merge conflict pains

gaetanmaisse commented on app/angular/package.json:
```diff
@@ -65,7 +65,7 @@
     "@babel/core": "*",
     "rxjs": "^6.0.0",
-    "typescript": "^3.4.0",
+    "typescript": "^3.9.3",
```
I'm not sure we should and need to bump this `peerDependencies` as it will create warnings when running `yarn/npm install` in a project using for instance Angular 6.

gaetanmaisse replied:
We have `"typescript": "^3.9.3",` in root `package.json` so we should be already using 3.9.

gaetanmaisse commented on addons/a11y/package.json:
```diff
@@ -61,9 +61,9 @@
   "gitHead": "57798a108e183d97f94fd4cc396e45e1d4b04d23",
   "typesVersions": {
-    "<=3.5": {
+    "<3.8": {
```
I think we can even go with `<3.9`

gaetanmaisse commented:
👋🏻  @hipstersmoothie I started to take a look, I will deep dive tomorrow and maybe push commits to your branch.

gaetanmaisse committed:
fix: put back `@babel/core` dep in `@storybook/core`

We are using `babel-loader` in `@storybook/core`.
And `babel-loader` has a peerDep on `@babel/core`.
So we have to add it to the dependencies of `@storybook/core`.

gaetanmaisse committed:
chore: bump `downlevel-dts` to ensure all TS 3.8+ features are properly downgraded to something understood by TS 3.4

gaetanmaisse committed:
refactor: rename downleveled TS directory to ts3.4

As of 0.6.0, `downlevel-dts` targets TypeScript 3.4 and there is no option to change that.
So I renamed all the output dirs to `ts3.4` to better represent what's inside them.

gaetanmaisse committed:
chore: update TypeScript peerDep version in @storybook/angular

`@storybook/angular` is compatible with old version of Angular.
However, these old versions only officially support some versions of TS.
So I reverted the update made in c8a60565b01f44659ecbd7af9ab7891b7e075456

gaetanmaisse committed:
Merge branch 'next' into deps

gaetanmaisse commented on addons/docs/package.json:
```diff
@@ -66,6 +66,7 @@
     "doctrine": "^3.0.0",
     "escodegen": "^1.12.0",
     "fast-deep-equal": "^3.1.1",
+    "glob": "^7.1.6",
```
Is this needed?

shilman replied:
shouldn't be

gaetanmaisse replied:
Removed it 🗑️

hipstersmoothie replied:
Oh that was for my other PR!

gaetanmaisse reviewed (approved):
LGTM, just a small comment on `glob` dep added in `@storybook/addon-docs`

gaetanmaisse committed:
chore: remove `glob` from deps of `@storybook/addon-docs`

`glob` was added in c8a60565b01f44659ecbd7af9ab7891b7e075456 but it looks there is no reason to do so 🤷‍.

shilman reviewed (approved):
Thanks for breaking this out @hipstersmoothie and getting it over the line @gaetanmaisse. You guys are awesome!

hipstersmoothie commented:
@gaetanmaisse @shilman thanks! I'll start updating the other PR
